src/services/upload.ts

```diff
@@ -0,0 +1,125 @@
+import { storage } from "~/firebase/init";
+import {
+  ref,
+  uploadBytes,
+  getDownloadURL,
+  FirebaseStorage,
+  listAll,
+  StorageReference,
+  uploadBytesResumable,
+  deleteObject,
+} from "firebase/storage";
+import { Dispatch } from "react";
+
+type UploadFilePath = {
+  userId: string;
+  file: File;
+};
+type UpdateFile = {
+  filePath: string;
+  file: File;
+};
+
+class UploadService {
+  public readonly rootFolder: string = "jazzapp/music/";
+
+  public async uploadFile(
+    fPath: UploadFilePath,
+    setProgress?: Dispatch<{ percents: number; operation: string }>,
+  ): Promise<{ status: boolean; url?: string }> {
+    if (!fPath.file) return Promise.resolve({ status: false });
+
+    const subMusicRef = `/${fPath.userId}/${fPath.file.name}`;
```

Comment:
Do not use `filename` here. As there might be two files with one name. Use uuid function:
```tsx
import { v4 as uuidv4 } from "uuid";

uuidv4();
```